mongo/with_transactions_test.go

```diff
@@ -381,6 +393,24 @@ func TestConvenientTransactions(t *testing.T) {
 		// Assert that transaction is canceled within 500ms and not 2 seconds.
 		assert.Soon(t, callback, 500*time.Millisecond)
 	})
+	t.Run("wrapped transient transaction error retried", func(t *testing.T) {
+		sess, err := client.StartSession()
+		assert.Nil(t, err, "StartSession error: %v", err)
+		defer sess.EndSession(context.Background())
+
+		returnError := true
```

Comment:
Can you add a comment describing what this is doing? It took me a couple reads to realize that `returnError` is being used to track whether the callback is executing only once or is being retried.